app/scripts/keyring-controller.js

```diff
@@ -273,6 +272,7 @@ module.exports = class KeyringController extends EventEmitter {
   setSelectedAccount (address) {
     var addr = normalize(address)
     this.configManager.setSelectedAccount(addr)
+    this.emit('update')
```

Comment:
This probably works, but we also have a more full way of doing this now.  It's in two parts:

Here, instead of these last two lines, we could just call `return this.fullUpdate()`.
In the `setSelectedAccount` action, we would then be able to take advantage of our new `callBackgroundThenUpdate()` method, which handles calling background methods like this, and then receiving a state update and using it.